ouroboros-consensus/src/Ouroboros/Consensus/Util/CBOR.hs

```diff
@@ -1,6 +1,7 @@
 {-# LANGUAGE RankNTypes          #-}
 {-# LANGUAGE RecordWildCards     #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns        #-}
 
 module Ouroboros.Consensus.Util.CBOR (
```

Comment:
@mrBliss @edsko Please take a look at this. I modified it so that the bytes from which the block or EBB was decoded are passed to the function to compute the `Maybe hash` for EBBs. It's needed in order to get that hash for the `cardano-ledger` boundary block type.